### DISCUSSION TITLE
Allow customizing the prefix for Dynamic Segments

### DISCUSSION THREAD

**Discussion by ianwremmel** [remix-run/remix#4021] [Proposals]
While using `$` as the dynamic segment indicators seems to make sense  (dollar sign being a common variable indicator), it _really_ makes scripting difficult. I'm using Remix in a few different projects that are orchestrated by Makefiles and there's really no way to escape the `$`. 

Would it be possible to allow providing an alternate character (or string) via `remix.config.js`?

**Comment by kiliman** [comment]
Here's a function that will let you choose any character as the param prefix.
```ts
// remix.config.js
const {
  defineConventionalRoutes,
} = require(https://github.com/remix-run/remix/tree/master/packages/remix-dev);

/**
 * @type {import(https://github.com/remix-run/remix/tree/master/packages/remix-dev).AppConfig}
 */
module.exports = {
  ignoredRouteFiles: ["**/.*"],
  routes: () => updateParamPrefix("^"),
};

function updateParamPrefix(paramPrefix) {
  const conventionalRoutes = defineConventionalRoutes("app", ["**/.*"]);

  const fixPath = (path) => {
    return path === paramPrefix
      ? "*" // handle catch-all
      : path?.replaceAll(paramPrefix, ":");
  };

  return Object.fromEntries(
    Object.entries(conventionalRoutes).map(([key, route]) => [
      key,
      { ...route, path: fixPath(route.path) },
    ])
  );
}
```
https://gist.github.com/kiliman/9d238aebd0fce58de20878803615f729

**Reply by ianwremmel** [reply]
oh, awsome, thanks!
